src/nvim/api/options.c

```diff
@@ -101,11 +103,19 @@ static int validate_option_value_args(Dict(option) *opts, char *name, OptIndex *
 }
 
 /// Create a dummy buffer and run the FileType autocmd on it.
-static buf_T *do_ft_buf(char *filetype, aco_save_T *aco, Error *err)
+static buf_T *do_ft_buf(const char *filetype, aco_save_T *aco, bool *aco_used, Error *err)
+  FUNC_ATTR_NONNULL_ARG(2, 3, 4)
 {
+  *aco_used = false;
   if (filetype == NULL) {
     return NULL;
   }
+  // Not firing nested FileType autocmds; bail early if one's running.
+  if (in_filetype_autocmd()) {
+    api_set_error(err, kErrorTypeException,
+                  "Cannot detect default while FileType autocommands are running");
```

Comment:
This issue bit me recently. `nvim_get_option_value` sometimes gave incorrect defaults from `LspAttach` when it indirectly ran from `FileType`...

Maybe some level of recursion can be allowed later, but for now the error is better than silence.

Reply:
After a bit of tinkering, I feel like it's _maybe_ better to just allow it and let the `apply_autocmds_group` nesting check take the wheel. :innocent: